source/common/upstream/cluster_manager_impl.cc

```diff
@@ -619,6 +635,7 @@ Http::AsyncClient& ClusterManagerImpl::httpAsyncClientForCluster(const std::stri
 ClusterUpdateCallbacksHandlePtr
 ClusterManagerImpl::addThreadLocalClusterUpdateCallbacks(ClusterUpdateCallbacks& cb) {
   ThreadLocalClusterManagerImpl& cluster_manager = tls_->getTyped<ThreadLocalClusterManagerImpl>();
+  cluster_manager.update_callbacks_.emplace_back(&cb);
```

Comment:
this is not related to this change. But I guess, cluster callbacks are missing. so added this.